tests/.infra/crw-ci/pr-check/k8s/Jenkinsfile

```diff
@@ -233,25 +218,39 @@ pipeline {
             }
 
             steps {
-                sh """
-                    ${WORKSPACE}/dockerfiles/che/build.sh --organization:eclipseche --tag:${mutableCheImageTag} --dockerfile:Dockerfile
-                    docker tag eclipseche/che-server:${mutableCheImageTag} docker.io/${cheImageRepo}:${mutableCheImageTag}
-                    docker push docker.io/${cheImageRepo}:${mutableCheImageTag}
-                """
+                withCredentials([string(credentialsId: 'ed71c034-60bc-4fb1-bfdf-9570209076b5', variable: 'maxura_docker_password')]) {
+                    sh """
+                        echo "Build 'che-server' image"
+                        echo "${maxura_docker_password}" | docker login -u maxura --password-stdin
+
+                        ${WORKSPACE}/dockerfiles/che/build.sh --organization:eclipseche --tag:${mutableCheImageTag} --dockerfile:Dockerfile
+                    """
+                }
+
+                withCredentials([string(credentialsId: 'quay.io-crw_pr+crw_pr_bot_user_token', variable: 'QUAY_TOKEN')]) {
+                    sh """
+                        echo "Push 'che-server' image"
+                        echo "${QUAY_TOKEN}" | docker login -u="crw_pr+crw_pr_bot" --password-stdin quay.io/crw_pr
+
+                        docker tag eclipseche/che-server:${mutableCheImageTag} ${cheImageRepo}:${mutableCheImageTag}
+                        echo y | docker push ${cheImageRepo}:${mutableCheImageTag}
+                    """
+                }
             }
         }
 
         stage("Start Che") {
             steps {
                 script {
-                    echo "Pre-pull Eclipse Che and test images from dockerhub"
+                    echo "Pre-pull test images"
                     withCredentials([string(credentialsId: 'ed71c034-60bc-4fb1-bfdf-9570209076b5', variable: 'maxura_docker_password')]) {
                         sh """
                             eval \$(minikube docker-env)
-                            docker login -u maxura -p ${maxura_docker_password}
 
                             docker pull ${cheImageRepo}:${mutableCheImageTag}
-                            docker pull docker.io/traefik:v2.2.8
+
+                            echo "${maxura_docker_password}" | docker login -u="maxura" --password-stdin docker.io
+                            docker pull docker.io/traefik:v2.2.8 || true
```

Comment:
Looks like we keep one dependency?

Reply:
Yes - it's needed to be able to run tests against Single Host Che